python/migrations/07_create_location_id_trigger/up.sql

```diff
@@ -0,0 +1,59 @@
+-- Add triggers to calculate location from latitude and longitude on crash insert and also crash edit update
+CREATE OR REPLACE FUNCTION cris.crash_create_location()
+RETURNS TRIGGER AS $$
+DECLARE
+    updated_location_id varchar;
+    updated_position geometry(Geometry,4326);
+    edited_latitude float;
+    edited_longitude float;
+BEGIN
+    -- Determine if the crash latitude and longitude have been edited
+    IF TG_OP = 'UPDATE' AND TG_TABLE_NAME = 'crash_cris_data' THEN
+        SELECT latitude, longitude INTO edited_latitude, edited_longitude FROM cris.crash_edit_data WHERE crash_id = NEW.crash_id;
+
+        IF edited_latitude IS NOT NULL AND edited_longitude IS NOT NULL THEN
```

Comment:
if only latitude or longitude has been edited by VZ wouldnt we still wanna give their edit priority in terms of calculating location id? like if CRIS put the wrong val for latitude so vz edited it, then for location ID we would use the combo of the vz latitude edit and the cris longitude?

Reply:
yep, that is a great catch. i was stuck on thinking about these two fields only being edited together through a map UI, but we do give VZ users the power to edit those values in the current form so it should work the way you describe. 🙏